specification/common-types/resource-management/v5/networksecurityperimeter.json

```diff
@@ -0,0 +1,400 @@
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the resource association",
+          "type": "string"
+        },
+        "accessMode": {
+          "$ref": "#/definitions/ResourceAssociationAccessMode"
+        }
+      }
+    },
+    "NetworkSecurityProfile": {
+      "description": "Network security perimeter configuration profile",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the profile",
+          "type": "string"
+        },
+        "accessRulesVersion": {
+          "description": "Current access rules version",
+          "type": "integer",
+          "format": "int32"
+        },
+        "accessRules": {
+          "description": "List of Access Rules",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AccessRule"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        }
+      }
+    },
+    "AccessRule": {
+      "description": "Access rule in a network security perimeter configuration profile",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the access rule",
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/AccessRuleProperties"
+        }
+      }
+    },
+    "AccessRuleDirection": {
+      "description": "Direction of Access Rule",
+      "enum": [
+        "Inbound",
+        "Outbound"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "AccessRuleDirection",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Inbound",
+            "description": "Applies to inbound network traffic to the secured resources."
+          },
+          {
+            "value": "Outbound",
+            "description": "Applies to outbound network traffic from the secured resources"
+          }
+        ]
+      }
+    },
+    "AccessRuleProperties": {
+      "description": "Properties of Access Rule",
+      "type": "object",
+      "properties": {
+        "direction": {
+          "$ref": "#/definitions/AccessRuleDirection"
+        },
+        "addressPrefixes": {
+          "description": "Address prefixes in the CIDR format for inbound rules",
+          "type": "array",
+          "items": {
+            "description": "An IP address prefix (CIDR) for inbound rules",
+            "type": "string"
+          }
+        },
+        "subscriptions": {
+          "description": "Subscriptions for inbound rules",
+          "type": "array",
+          "items": {
+            "description": "Subscription identifiers",
+            "type": "object",
+            "properties": {
+              "id": {
+                "description": "The fully qualified Azure resource ID of the subscription e.g. ('/subscriptions/00000000-0000-0000-0000-000000000000') ",
+                "type": "string"
```

Comment:
Shouldn't this be format arm-id?

Reply:
I guess you're right, it should, in fact. I don't see it used for subscriptions very often, let's see if it works!

Reply:
Yeah, for built-in resources, it may not.